### DISCUSSION TITLE
Can theia extension get own public folder?

### DISCUSSION THREAD

**Discussion by saltylassi** [eclipse-theia/theia#11098] [Q&A] [answered]
Hello theia,

I'm making an extension handling response from external server , and I got errors.

response is string type html, and my extension parses that string to use widget contents like customEditor.
<br/><br/><br/>

**Here is the question :** 

response has relative path like ```<link rel="stylesheet" href="./_resource_/main.css">```.

When I parse this, theia understands path like ```browser-app/lib/_resource_/main.css```.

How can I use custom static folder on each extension?

I think this is normal behavior because theia uses expressJS, but there is any solution about this?
<br/><br/><br/>

※ I'm using widget extension sample :)

https://theia-ide.org/docs/widgets/

**Comment by msujew** [comment, marked as answer]
Hi @saltylassi,

You can create a custom `BackendApplicationContribution` to add a new route to the express app. Something like this:

```ts
@injectable()
export class CustomBackendContribution implements BackendApplicationContribution {

    configure(app: express.Application): void {
        app.get('/_resource_/:file', async (req, res) => {
            const file = req.params.file;
            const fileContent = fs.readFile(path.join(__dirname, '..', file), 'utf8');
            res.send(fileContent);
        });
    }

}
```
